### PR TITLE
Md5 check failed

### PR DESCRIPTION
**Pull request by jermnelson**
## Why was this change made?
Creates a new error to record if a file in the Google Books download fails a MD5 check.

Part of #93 

## Was the documentation (README, DevOpsDocs, wiki, openapi.json) updated?
n/a

### CODE REVIEW

File: spec/models/book_spec.rb [hunk 1]
```diff
@@ -92,4 +92,16 @@
       expect(book.last_download_failed_reason).to eq(reason)
     end
   end
+
+  describe '#md5_check_failed' do
+    let(:reason) { 'file foo.txt MD5 in test/checksum.md5 does not match hash abcd358979' }
+
+    it('updates status, last_download_date, and last_process_failed_reason') do
+      puts book
```
Comment (jcoyne) [inline spec/models/book_spec.rb]: Can this be removed?

Comment (jermnelson) [reply spec/models/book_spec.rb]: Done.

**Comment by jermnelson** [comment]
> Looks like this also needs a rebase.

Done.

File: spec/models/book_spec.rb [hunk 2]
```diff
@@ -114,4 +114,15 @@
...
   end
+
+  describe '#md5_check_failed' do
+    let(:reason) { 'file foo.txt MD5 in test/checksum.md5 does not match hash abcd358979' }
+
+    it('updates status, last_download_date, and last_process_failed_reason') do
+      book.md5_check_failed(status_date, reason)
+      expect(book.status).to eq(Book::MD5_CHECK_FAILED)
+      expect(book.last_download_date).to eq(status_date)
+      # expect(book.last_process_failed_reason).to eq(reason)
```
Comment (aaron-collier) [inline spec/models/book_spec.rb]: Cruft?

Comment (jermnelson) [reply spec/models/book_spec.rb]: Fixed; uncommented this `expect` statement.

File: db/migrate/20200113221026_book.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Book < ActiveRecord::Migration[6.0]
+  def change
```
Comment (jcoyne) [inline db/migrate/20200113221026_book.rb]: Why do we have an empty migration? Can this be removed?

File: db/schema.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2019_04_23_123605) do
+ActiveRecord::Schema.define(version: 2020_01_13_221026) do
```
Comment (jcoyne) [inline db/schema.rb]: and revert this?

File: db/migrate/20200113221026_book.rb (hunk 3 above)
Comment (jermnelson) [reply db/migrate/20200113221026_book.rb]: Removed and changed pushed up.

File: db/schema.rb (hunk 4 above)
Comment (jermnelson) [reply db/schema.rb]: Done and pushed up.

**Pull request merged**